### PR TITLE
Add schema component

### PR DESCRIPTION
**Pull request by tahmidefaz**
Adding a section to show the schema.
A few more information needs to be loaded and the styling needs to be improved, but those can be addressed in a separate PR.

image

### CODE REVIEW

**Comment by tahmidefaz** [comment]
The double `value` in the image is caused by an `oneOf`  section. We have a ticket open to handle that property.

File: src/components/APIDoc/ApiDoc.tsx [hunk 1]
```diff
@@ -51,5 +52,10 @@ export const ApiDoc: React.FunctionComponent<ApiDocProps> = props => {
             </StackItem>
         )}
         { paths.loading ? <Bullseye><Spinner /></Bullseye> : paths.value }
+        { paths.loading ? <Bullseye><Spinner /></Bullseye>:
+            <StackItem>
+                <SchemaViewer document={ openapi }/>
+            </StackItem>
+        }
```
Comment (josejulio) [inline src/components/APIDoc/ApiDoc.tsx:59]: The paths.loading stuff was used to generate the markup in the "background".

```suggestion
        <StackItem>
            <SchemaViewer document={ openapi }/>
        </StackItem>
```

Comment (josejulio) [reply src/components/APIDoc/ApiDoc.tsx:59]: We should delete the `const paths` above as well.

File: src/components/APIDoc/Operation.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import {OpenAPIV3} from "openapi-types";
-import {buildExample, deRef} from "../../utils/Openapi";
+import { buildExample, deRef } from '../../utils/Openapi';
```
Comment (josejulio) [inline src/components/APIDoc/Operation.tsx:3]: I guess we should add a linter to unify this - I like the space in the braces and using `'` insteasd of `"` - but lets try to have some tooling to prevent from accepting different code. 

Meanwhile I suggest to remove this change
```suggestion
import {buildExample, deRef} from "../../utils/Openapi";
```

File: src/components/APIDoc/ApiDoc.tsx (hunk 1 above)
Comment (josejulio) [reply src/components/APIDoc/ApiDoc.tsx:59]: Ah - hold on - I guess i got something wrong

File: src/components/APIDoc/Operation.tsx (hunk 2 above)
Comment (tahmidefaz) [reply src/components/APIDoc/Operation.tsx:3]: Ah, I think I ran `npm run build` before pushing. That might have made that unrelated change. Not sure. :thinking:

File: src/components/APIDoc/ApiDoc.tsx (hunk 1 above)
Comment (josejulio) [reply src/components/APIDoc/ApiDoc.tsx:59]: OK - so the schema viewer is the part that shows the types - My initial suggestion still applies

```suggestion
        <StackItem>
            <SchemaViewer document={ openapi }/>
        </StackItem>
```

File: src/components/APIDoc/SchemaDataView.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  if (!schema.properties) {
+    return [{name: "schema undefined"}] as TreeViewDataItem[]
+  }
+  const schemaData = Object.entries(schema.properties).map(([key, value]) => {
+    let propertyType: string
+
+    if ('type' in value) {
+      propertyType = value.type as string
```
Comment (josejulio) [inline src/components/APIDoc/SchemaDataView.tsx:71]: We also need to check for `enum` - but we can do it on a separate PR. enum will restrict the values to specific instances of the type.

**Review by josejulio — APPROVED** [review]
lgtm, some minor comments

File: src/components/APIDoc/SchemaDataView.tsx (hunk 3 above)
Comment (tahmidefaz) [reply src/components/APIDoc/SchemaDataView.tsx:71]: Yeah, we will need to do another pass on this to include all the optional info like `enum`, `max`, `min` etc.

File: src/components/APIDoc/SchemaDataView.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    } else if ('$ref' in value ) {
+      propertyType = value.$ref.split('/').at(-1) as string
```
[the remark this answers is not in the document]
Comment (tahmidefaz) [reply src/components/APIDoc/SchemaDataView.tsx:73]: I tried it, and this might not work in this section, as some of these `refs` are actually one level deeper than the `deRefData` of the schema. We will need to call `deRef` here again if we want to use `deRefData.name` at this section.
